app/services/organization_update_service.rb

```diff
@@ -1,27 +1,54 @@
 class OrganizationUpdateService
-  # @param organization [Organization]
-  # @param params [ActionDispatch::Http::Parameters]
-  # @return [Boolean]
-  def self.update(organization, params)
-    result = organization.update(params)
-    return false unless result
-
-    update_partner_flags(organization)
-    true
-  end
-
-  # @param organization [Organization]
-  def self.update_partner_flags(organization)
-    fields = %i[
+  class << self
+    FIELDS = %i[
       enable_child_based_requests
       enable_individual_requests
       enable_quantity_based_requests
     ]
-    fields.each do |field|
-      next unless organization.saved_change_to_attribute?(field)
-      organization.partners.map(&:profile).each do |profile|
-        profile.update!(field => organization.send(field))
+
+    # @param organization [Organization]
+    # @param params [ActionDispatch::Http::Parameters]
+    # @return [Boolean]
+    def update(organization, params)
+      return false unless valid?(organization, params)
+
+      result = organization.update(params)
+      return false unless result
+
+      update_partner_flags(organization)
+      true
+    end
+
+    # @param organization [Organization]
+    def update_partner_flags(organization)
+      FIELDS.each do |field|
+        # We don't want to automatically enable request types
+        # on a partner. This should be left up to
+        # individual partners to decide themselves as per
+        # github.com/rubyforgood/human-essentials/issues/3264
+        next if organization.send(field)
+        organization.partners.map(&:profile).each do |profile|
```

Comment:
This should be `each` rather than `map` since you're not doing anything with the result of the block.

Reply:
The result of the block returns an array of profiles, and then calling `.each` on that array allows us to loop through all the profiles.

If I swap the `map` with `each` it throws an error - unless you meant something else?

Reply:
Whoops! Ignore me, I was reading the code incorrectly 🙈